openeuler/openeuler_test.go

```diff
@@ -0,0 +1,151 @@
+	}{
+		{
+			name:  "positive test openEuler",
+			appFs: afero.NewMemMapFs(),
+			xmlFileNames: map[string]string{
+				"/security/data/cvrf/index.txt":                            "testdata/index.txt",
+				"/security/data/cvrf/2021/cvrf-openEuler-SA-2021-1033.xml": "testdata/cvrf-openEuler-SA-2021-1033.xml",
+				"/security/data/cvrf/2023/cvrf-openEuler-SA-2023-1374.xml": "testdata/cvrf-openEuler-SA-2023-1374.xml",
+				"/security/data/cvrf/2024/cvrf-openEuler-SA-2024-1349.xml": "testdata/cvrf-openEuler-SA-2024-1349.xml",
+				"/security/data/cvrf/2022/cvrf-openEuler-SA-2022-1693.xml": "testdata/cvrf-openEuler-SA-2022-1693.xml",
+			},
+			goldenFiles: map[string]string{
+				"/tmp/cvrf/openeuler/2021/openEuler-SA-2021-1033.json": "testdata/golden/openEuler-SA-2021-1033.json",
+				"/tmp/cvrf/openeuler/2023/openEuler-SA-2023-1374.json": "testdata/golden/openEuler-SA-2023-1374.json",
+				"/tmp/cvrf/openeuler/2024/openEuler-SA-2024-1349.json": "testdata/golden/openEuler-SA-2024-1349.json",
+				"/tmp/cvrf/openeuler/2022/openEuler-SA-2022-1693.json": "testdata/golden/openEuler-SA-2022-1693.json",
+			},
+		},
+		{
+			name:  "invalid filesystem write read only path",
+			appFs: afero.NewReadOnlyFs(afero.NewOsFs()),
+			xmlFileNames: map[string]string{
+				"/security/data/cvrf/index.txt":                            "testdata/invalid-index.txt",
+				"/security/data/cvrf/2021/cvrf-openEuler-SA-2021-1202.xml": "testdata/golden/openEuler-SA-2021-1202.json",
+			},
+			goldenFiles:      map[string]string{},
+			expectedErrorMsg: "failed Update openEuler CVRF: failed to decode openEuler cvrf XML: EOF",
+		},
+		{
+			name:  "empty file format",
+			appFs: afero.NewMemMapFs(),
+			xmlFileNames: map[string]string{
+				"/security/data/cvrf/index.txt":                            "testdata/invalid-index.txt",
+				"/security/data/cvrf/2021/cvrf-openEuler-SA-2021-1202.xml": "testdata/EOF.txt",
+			},
+			goldenFiles:      map[string]string{},
+			expectedErrorMsg: "",
+		},
+		{
+			name:  "invalid file format",
+			appFs: afero.NewMemMapFs(),
+			xmlFileNames: map[string]string{
+				"/security/data/cvrf/index.txt":                            "testdata/invalid-index.txt",
+				"/security/data/cvrf/2021/cvrf-openEuler-SA-2021-1202.xml": "testdata/invalid.txt",
+			},
+			goldenFiles:      map[string]string{},
+			expectedErrorMsg: "failed Update openEuler CVRF: failed to decode openEuler cvrf XML: EOF",
+		},
+		{
+			name:  "broken XML",
+			appFs: afero.NewMemMapFs(),
+			xmlFileNames: map[string]string{
+				"/security/data/cvrf/index.txt":                            "testdata/invalid-index.txt",
+				"/security/data/cvrf/2021/cvrf-openEuler-SA-2021-1202.xml": "testdata/broken-cvrf.xml",
+			},
+			goldenFiles:      map[string]string{},
+			expectedErrorMsg: "failed Update openEuler CVRF: failed to decode openEuler cvrf XML: XML syntax error on line 180: unexpected EOF",
+		},
+	}
+	for _, tc := range testCases {
+		t.Log("Entering...")
```

Comment:
I think we can delete this log.
`Go` says to start a test case:
```
=== RUN   TestConfig_Update/positive_test_openEuler
    openeuler_test.go:87: http ready to start...
    openeuler_test.go:99: http started!
    openeuler_test.go:108: updating...
```

Reply:
funny case:
i run only 1 testcase, but still see this log for all testcases:
```
=== RUN   TestConfig_Update
    openeuler_test.go:86: Entering...
    openeuler_test.go:86: Entering...
    openeuler_test.go:86: Entering...
    openeuler_test.go:86: Entering...
    openeuler_test.go:86: Entering...
--- FAIL: TestConfig_Update (0.04s)
=== RUN   TestConfig_Update/positive_test_openEuler
    openeuler_test.go:88: http ready to start...
    openeuler_test.go:100: http started!
    openeuler_test.go:109: updating...
```

Better doesn't add some code before `t.Run`